File: ValorPalabra.java
import java.util.Scanner;

public class ValorPalabra {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int total;

        do {
            total = 0;
            System.out.println("Ingrese una palabra:");
            String palabra = sc.nextLine();

            for (int i = 0; i < palabra.length(); i++) {
                char c = Character.toLowerCase(palabra.charAt(i));
                if (c >= 'a' && c <= 'z')
                    total += (c - 'a' + 1);
            }

            System.out.println("Valor de la palabra: " + total);
        } while (total != 100);

        System.out.println("Programa finalizado (palabra con 100 puntos).");
    }
}

File: valor_palabra.f
        program valor_palabra
        implicit none
        character(len=100) :: palabra
        integer :: i, total, longitud

        do
          print *, "Ingrese una palabra:"
          read(*,'(A)') palabra
          total = 0
          longitud = len_trim(palabra)

            do i = 1, longitud
            if (palabra(i:i) >= 'A' .and. palabra(i:i) <= 'Z') then
                total = total + iachar(palabra(i:i)) - iachar('A') + 1
            else if (palabra(i:i) >= 'a' .and. palabra(i:i) <= 'z') then
                total = total + iachar(palabra(i:i)) - iachar('a') + 1
            end if
            end do

          print *, "Valor de la palabra:", total

          if (total == 100) exit
          end do

          print *, "Programa finalizado (palabra con 100 puntos)."
          PRINT *
          PRINT *, 'Presione ENTER para salir...'
          READ(*,*)
          
          end program valor_palabra

